File: abc/abc125/abc125_d.cs
// https://atcoder.jp/contests/abc125/submissions/14522750

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = Console.ReadLine();
        var A = Console.ReadLine().Split().Select(long.Parse).ToArray();
        var AA = A.Select(x => Math.Abs(x));

        Console.WriteLine(A.Any(x => x == 0) || A.Count(x => x < 0) % 2 == 0 ? AA.Sum() : AA.Sum() - 2 * AA.Min());
    }
}

File: abc/abc125/abc125_d.fs
// https://atcoder.jp/contests/abc125/submissions/14523283

let N = stdin.ReadLine()
let A = stdin.ReadLine().Split() |> Array.map int64
let AA = Array.map abs A

if A
   |> Array.contains 0L
   || (A
       |> Array.filter (fun x -> x < 0L)
       |> Array.length) % 2 = 0
then Array.sum AA
else (Array.sum AA) - 2L * (Array.min AA)
|> stdout.WriteLine
